src/Kestrel.Core/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -35,6 +35,11 @@ public void Abort(Exception ex)
         {
             lock (_writeLock)
             {
+                if (_completed)
```

Comment:
Is not better use double-checked locking ?
like this:

```
if(!_completed)
{
      lock(_writeLock)
      {
            if(!_completed)
            {
                    _outputReader.CancelPendingRead();
                    _outputWriter.Complete(ex);
                   _completed = true;
            }
      }
}
```

Reply:
Likely only ever called once if at all, so wouldn't make much difference?

Reply:
Likely only ever called once if at all, so wouldn't make much difference?

Also the second call would complete before its finished as it wouldn't wait for the lock

Reply:
I don't know, I suggested that code because I saw lock and  double-checked locking can be used in this situation , but if is only ever called once maybe doesn't make much difference, OK

Reply:
It's not on a hot path, so unlikely to benefit from double-checked locking.